### PR TITLE
wip: update to the recent x version

### PR DESCRIPTION
**Pull request by dadrus**
@zepatrik: This the work in progress PR, you helped me today with.

## Related issue(s)

None I'm aware of. 

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I am following the [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security. vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

## Further Comments

Work in progress. ;)

Current status: Does not compile as following needs still to be fixed:
```
github.com/ory/oathkeeper/driver/health
../driver/health/event_manager_default.go:64:27: cannot use listener.Validate (type func() error) as type healthx.ReadyChecker in assignment

github.com/ory/oathkeeper/driver/configuration
../driver/configuration/provider_viper.go:332:28: undefined: wiperx
```

### CODE REVIEW

**Comment by dadrus** [comment]
@zepatrik: It would be great if you could take a look at `event_manager_default.go:64:27`. I'll try to resolve the other one by myself

File: pipeline/mutate/mutator_hydrator.go [hunk 1]
```diff
@@ -184,26 +188,31 @@ func (a *MutatorHydrator) Mutate(r *http.Request, session *authn.AuthenticationS
 	}
 	req.Header.Set(contentTypeHeaderKey, contentTypeJSONHeaderValue)
 
-	var client http.Client
+	var client *retryablehttp.Client
 	if cfg.Api.Retry != nil {
 		maxRetryDelay := time.Second
-		giveUpAfter := time.Millisecond * 50
+		// giveUpAfter := time.Millisecond * 50 this was previously used as maxelapsedtime for the old transport implementation
```
Comment (dadrus) [inline pipeline/mutate/mutator_hydrator.go:194]: @aeneasr : Can you please help with this one?

File: pipeline/mutate/mutator_hydrator.go [hunk 2]
```diff
@@ -184,26 +188,31 @@ func (a *MutatorHydrator) Mutate(r *http.Request, session *authn.AuthenticationS
...
-		if len(cfg.Api.Retry.GiveUpAfter) > 0 {
-			if d, err := time.ParseDuration(cfg.Api.Retry.GiveUpAfter); err != nil {
-				a.d.Logger().WithError(err).Warn("Unable to parse max_delay in the Hydrator Mutator, falling pack to default.")
-			} else {
-				giveUpAfter = d
-			}
-		}
 
-		client.Transport = httpx.NewResilientRoundTripper(a.client.Transport, maxRetryDelay, giveUpAfter)
+		// TODO: for the below block (related to the todo above)
```
Comment (dadrus) [inline pipeline/mutate/mutator_hydrator.go:204]: @aeneasr : and with this one please ;)

File: pipeline/mutate/mutator_hydrator.go (hunk 1 above)
Comment (aeneasr) [reply pipeline/mutate/mutator_hydrator.go:194]: You mean whether this default is ok?

File: pipeline/mutate/mutator_hydrator.go (hunk 2 above)
Comment (aeneasr) [reply pipeline/mutate/mutator_hydrator.go:204]: Not sure what you mean exactly 😅

**Review by zepatrik — COMMENTED** [review]
Hm no idea, sorry. Maybe if you bump to ory/x v0.0.356 the dependency is actually gone? I was able to bring down the number of dependencies quite a lot there.

**Comment by aeneasr** [comment]
Sorry for replying so late. This is the root cause of the problem: https://github.com/dgraph-io/ristretto/pull/292

I fixed this with this rewrite:

```
replace github.com/dgraph-io/ristretto => github.com/ory/ristretto fix-log
```

**Comment by aeneasr** [comment]
Superseded by #999

**Pull request closed without merging**